chat-plugins/othermetas.js

```diff
@@ -147,6 +147,109 @@ exports.commands = {
+			baseTemplate = Dex.getTemplate("Groudon");
+			megaTemplate = Dex.getTemplate("Groudon-Primal");
+		} else if (stone.id === 'blueorb') {
+			baseTemplate = Dex.getTemplate("Kyogre");
+			megaTemplate = Dex.getTemplate("Kyogre-Primal");
+		}
+		let deltas = {
+			baseStats: {},
+			weightkg: megaTemplate.weightkg - baseTemplate.weightkg,
+		};
+		for (let statId in megaTemplate.baseStats) {
+			deltas.baseStats[statId] = megaTemplate.baseStats[statId] - baseTemplate.baseStats[statId];
+		}
+		if (megaTemplate.types.length > baseTemplate.types.length) {
+			deltas.type = megaTemplate.types[1];
+		} else if (megaTemplate.types.length < baseTemplate.types.length) {
+			deltas.type = baseTemplate.types[0];
+		} else if (megaTemplate.types[1] !== baseTemplate.types[1]) {
+			deltas.type = megaTemplate.types[1];
+		}
+		let details = {
+			"Gen": 6,
+			"Weight": (JSON.stringify(deltas.weightkg).startsWith("-") ? "" : "+") + Math.round(deltas.weightkg * 100) / 100 + " kg",
+		};
+		let tier;
+		if (['redorb', 'blueorb'].includes(stone.id)) {
+			tier = "Orb";
+		} else if (targetid === "dragonascent") {
+			tier = "Move";
+		} else {
+			tier = "Stone";
+		}
+		let buf = `<li class="result">`;
+		buf += `<span class="col numcol">${tier}</span> `;
+		if (targetid === "dragonascent") {
+			buf += `<span class="col itemiconcol"></span>`;
+		} else {
+			buf += `<span class="col itemiconcol"><psicon item="${targetid}"/></span> `;
+		}
+		if (targetid === "dragonascent") {
+			buf += `<span class="col movenamecol" style="white-space:nowrap"><a href="https://pokemonshowdown.com/dex/moves/${targetid}" target="_blank">Dragon Ascent</a></span> `;
+		} else {
+			buf += `<span class="col pokemonnamecol" style="white-space:nowrap"><a href="https://pokemonshowdown.com/dex/items/${stone.id}" target="_blank">${stone.name}</a></span> `;
+		}
+		if (deltas.type) {
+			buf += `<span class="col typecol"><img src="https://play.pokemonshowdown.com/sprites/types/${deltas.type}.png" alt="${deltas.type}" height="14" width="32"></span> `;
+		} else {
+			buf += `<span class="col typecol"></span>`;
+		}
+		buf += `<span style="float:left;min-height:26px">`;
+		buf += `<span class="col abilitycol">${megaTemplate.abilities['0']}</span>`;
+		buf += `<span class="col abilitycol"></span>`;
+		buf += `</span>`;
+		buf += `<span style="float:left;min-height:26px">`;
+		buf += `<span class="col statcol"><em>HP</em><br />0</span> `;
+		buf += `<span class="col statcol"><em>Atk</em><br />${deltas.baseStats.atk}</span> `;
+		buf += `<span class="col statcol"><em>Def</em><br />${deltas.baseStats.def}</span> `;
+		buf += `<span class="col statcol"><em>SpA</em><br />${deltas.baseStats.spa}</span> `;
+		buf += `<span class="col statcol"><em>SpD</em><br />${deltas.baseStats.spd}</span> `;
+		buf += `<span class="col statcol"><em>Spe</em><br />${deltas.baseStats.spe}</span> `;
+		buf += `<span class="col bstcol"><em>BST<br />100</em></span> `;
+		buf += `</span>`;
+		buf += `</li>`;
```

Comment:
Yes, this is annoying. At one point I was toying with the idea of having a single function in Chat which would look for well-known properties of the effect passed in and generate appropriate details. I wonder whether I should dig it out and update it to work with a set of mega deltas...